File: Data-Mining/assignment4/spectral_clustering/cluster_plot.m
function cluster_plot(X, idx)
%CLUSTER_PLOT show clustering results
%
%   Input: X   - data point features, n-by-p maxtirx.
%          idx - data point cluster labels, n-by-1 vector.


plot(X(idx==1,1),X(idx==1,2),'r.','MarkerSize',12)
hold on
plot(X(idx==2,1),X(idx==2,2),'b.','MarkerSize',12)

legend('Cluster 1','Cluster 2')
axis equal
end

File: Data-Mining/assignment4/spectral_clustering/cluster_plot.py
import matplotlib.pyplot as plt

def cluster_plot(X, idx):
    '''
    CLUSTER_PLOT show clustering results
    Input: 
        X - data point features, n-by-p maxtirx.
        idx - data point cluster labels, n-by-1 vector.
    '''
    plt.figure(figsize=(6, 6))
    plt.plot(X[idx==0, 0],X[idx==0,1],'r.',markersize=5, label='Cluster 1')
    plt.plot(X[idx==1, 0],X[idx==1,1],'b.', markersize=5, label='Cluster 2')
    plt.legend(loc='upper right')